File: cpp/Garage.cpp
#include <iostream>
#include "Garage.hpp"

// Constructor
Garage::Garage(const std::string &name, int sizeArea)
    : name(name), sizeArea(sizeArea) {
}

// Destructor
Garage::~Garage() {
    // Destructor content, if needed
}

// Getters
std::string Garage::getName() const {
    return name;
}

int Garage::getSizeArea() const {
    return sizeArea;
}

std::vector<Car>& Garage::getCars() {
    return listOfCar;
}

std::vector<Motorcycle>& Garage::getMotorcycles() {
    return listOfMotorcycle;
}

// Setters
void Garage::setName(const std::string &name) {
    this->name = name;
}

void Garage::setSizeArea(int sizeArea) {
    this->sizeArea = sizeArea;
}

void Garage::addCar(const Car &car) {
    listOfCar.push_back(car);
}

void Garage::addMotorcycle(const Motorcycle &motorcycle) {
    listOfMotorcycle.push_back(motorcycle);
}

void Garage::print() {
    std::cout << name << std::endl;

    std::cout << "\nList of number plate of car\n";
    for(Car car: listOfCar) {
        std::cout << car.getNumberPlate() << std::endl;
    }

    std::cout << "\nList of number plate of motorcyle\n";
    for(Motorcycle motorcycle: listOfMotorcycle) {
        std::cout << motorcycle.getNumberPlate() << std::endl;
    }
    
    std::cout << std::endl;
}

File: java/Garage.java
import java.util.ArrayList;
import java.util.List;

public class Garage {
    private String name;
    private int sizeArea;
    private List<Car> listOfCar;
    private List<Motorcycle> listOfMotorcycle;

    // Constructor
    public Garage(String name, int sizeArea) {
        this.name = name;
        this.sizeArea = sizeArea;
        this.listOfCar = new ArrayList<>();
        this.listOfMotorcycle = new ArrayList<>();
    }

    // Getters
    public String getName() {
        return name;
    }

    public int getSizeArea() {
        return sizeArea;
    }

    public List<Car> getCars() {
        return listOfCar;
    }

    public List<Motorcycle> getMotorcycles() {
        return listOfMotorcycle;
    }

    // Setters
    public void setName(String name) {
        this.name = name;
    }

    public void setSizeArea(int sizeArea) {
        this.sizeArea = sizeArea;
    }

    public void addCar(Car car) {
        listOfCar.add(car);
    }

    public void addMotorcycle(Motorcycle motorcycle) {
        listOfMotorcycle.add(motorcycle);
    }

    public void print() {
        System.out.println(name);

        System.out.println("\nList of number plate of car");
        for (Car car : listOfCar) {
            System.out.println(car.getNumberPlate());
        }

        System.out.println("\nList of number plate of motorcycle");
        for (Motorcycle motorcycle : listOfMotorcycle) {
            System.out.println(motorcycle.getNumberPlate());
        }

        System.out.println();
    }
}
